processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -438,6 +434,49 @@ private Assignment assignToPlainViaSetter(Type targetType, Assignment rhs) {
             }
         }
 
+        /**
+         * Checks whether the setter wrapper should include a null / presence check or not
+         *
+         * @param rhs the source right hand side
+         * @param targetType the target type
+         *
+         * @return whether to include a null / presence check or not
+         */
+        private boolean setterWrapperNeedsSourceNullCheck(Assignment rhs, Type targetType) {
+            if ( rhs.getSourceType().isPrimitive() && rhs.getSourcePresenceCheckerReference() == null ) {
+                // If the source type is primitive or it doesn't have a presence checker then
+                // we shouldn't do a null check
+                return false;
+            }
+
+            if ( nvcs == ALWAYS ) {
+                // NullValueCheckStrategy is ALWAYS -> do a null check
```

Comment:
This is a strange one.. NullValueCheckStrategy does concern parameters, not bean properties.. Do you remember why this one is here? Perhaps it requires some more comment 😄

Reply:
I think that you are talking about `NullValueMappingStrategy`. The `NullValueCheckStrategy` is for properties